controllers/pxc/transition_key_copier.go

```diff
@@ -0,0 +1,106 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	pxcv1 "github.com/Percona-Lab/k8s-vault-issuer/apis/pxc/v1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (r *PerconaXtraDBClusterReconciler) processTransferVaultKeys(currClusterCR *pxcv1.PerconaXtraDBCluster, clusters []string) error {
+	currClusterName := currClusterCR.Name + "." + currClusterCR.Namespace
+	for _, v := range clusters {
+		splittedName := strings.Split(v, ".")
+		if len(splittedName) != 2 {
+			return errors.New("Invalid source cluster name, plaease use format clusterName.namespace")
+		}
+
+		clusterName, clusterNamespace := splittedName[0], splittedName[1]
+
+		srcClusterCR := &pxcv1.PerconaXtraDBCluster{}
+		err := r.Client.Get(context.TODO(),
+			types.NamespacedName{
+				Namespace: clusterNamespace,
+				Name:      clusterName,
+			}, srcClusterCR)
+		if err != nil {
+			return errors.Wrap(err, "get cluster srcClusterCR from namespace "+v)
+		}
+
+		values, ok := srcClusterCR.Annotations["percona.com/allow-transition-key-transfer"]
+		if !ok {
+			return fmt.Errorf("No percona.com/allow-transition-key-transfer annotation found at cluster %s", v)
+		}
+
+		allowed := false
+		for _, allowedClusters := range strings.Split(values, ",") {
+			if allowedClusters == currClusterName {
+				allowed = true
+				break
+			}
+		}
+
+		if !allowed {
+			return fmt.Errorf("Cluster %s is not allowing copying keys for cluster %s ", v, currClusterName)
+		}
+
+		if err := r.transferTransitionKeys(srcClusterCR, currClusterCR); err != nil {
+			return errors.Wrapf(err, "transfer transition key from:%s to:%s", v, currClusterCR.Namespace)
+		}
+	}
+	return r.deleteAnnotation(currClusterCR, "percona.com/vault-transfer-keys")
```

Comment:
`return errors.Wrapf(err, "transfer transition key from:%s to:%s", v, currClusterCR.Namespace)`

Reply:
it's the same now